core/include/tile/tile.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file   tile.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2017 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class Tile.
+ */
+
+#ifndef TILEDB_TILE_H
+#define TILEDB_TILE_H
+
+#include "const_buffer.h"
+#include "status.h"
+
+#include <cinttypes>
+
+namespace tiledb {
+
+class Tile {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  Tile();
+
+  Tile(uint64_t tile_size);
+
+  ~Tile();
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+
+  void* buffer() const;
+
+  uint64_t buffer_size() const;
+
+  bool full() const;
+
+  void reset();
+
+  Status write(ConstBuffer* buf);
+
+ private:
+  /* ********************************* */
+  /*         PRIVATE ATTRIBUTES        */
+  /* ********************************* */
+
+  void* buffer_;
```

Comment:
why not use the above Buffer class here.  The buffer class can also have an offset (is this the position in the buffer for seeking?)